### PR TITLE
remove upper bound on network

### PR DESCRIPTION
**Pull request by larskuhtz**
Previous version of network the package (>= 3.1.1.0) had a bug that could cause nodes to stall after running for several days. The issue is reported here: https://github.com/haskell/network/issues/438.

Version [3.1.2.0 of the network package](https://github.com/haskell/network/releases/tag/v3.1.2.0) is supposed to fix this and we haven't observed the issue while testing this branch, which indicates that the issue is indeed fixed.

### CODE REVIEW

File: chainweb.cabal [hunk 1]
```diff
@@ -360,7 +360,7 @@ library
         , mtl >= 2.2
         , mwc-random >= 0.13
         , mwc-probability >= 2.0 && <2.4
-        , network >= 2.6 && < 3.1.1
+        , network >= 2.6
```
Comment (larskuhtz) [inline chainweb.cabal:363]:
```suggestion
        , network >= 2.6 && < 3.1.1 || >= 3.1.2
```

File: chainweb.cabal [hunk 2]
```diff
@@ -531,7 +531,7 @@ test-suite chainweb-tests
         , lens-aeson >= 1.1
         , loglevel >= 0.1
         , mtl >= 2.2
-        , network >= 2.6 && < 3.1.1
+        , network >= 2.6
```
Comment (larskuhtz) [inline chainweb.cabal:534]:
```suggestion
        , network >= 2.6 && < 3.1.1 || >= 3.1.2
```

File: chainweb.cabal [hunk 3]
```diff
@@ -696,7 +696,7 @@ executable cwtool
         , memory >=0.14
         , mtl >= 2.2
         , mwc-random >= 0.14
-        , network >= 2.6 && < 3.1.1
+        , network >= 2.6
```
Comment (larskuhtz) [inline chainweb.cabal:699]:
```suggestion
        , network >= 2.6 && < 3.1.1 || >= 3.1.2
```

**Comment by larskuhtz** [comment]
I have been running this code for over two weeks now on several nodes without any issue.

**Review by slpopejoy — APPROVED** [review]

**Pull request merged**